XamarinCommunityToolkit/Converters/ListIsNotNullOrEmptyConverter.shared.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Collections;
+using System.Globalization;
+using Xamarin.Forms;
+using System.Linq;
+
+namespace XamarinCommunityToolkit.Converters
+{
+    /// <summary>
+    /// Converts the incoming value to a boolean indicating whether or not the value is not null and not empty.
+    /// </summary>
+    public class ListIsNotNullOrEmptyConverter : IValueConverter
+    {
+        /// <summary>
+        /// Converts the incoming value to a boolean indicating whether or not the value is not null and not empty.
+        /// </summary>
+        /// <param name="value">The value to convert.</param>
+        /// <param name="targetType">The type of the binding target property.</param>
+        /// <param name="parameter">Additional parameter for the converter to handle. Not implemented.</param>
+        /// <param name="culture">The culture to use in the converter.</param>
+        /// <returns>A boolean indicating if the incoming value is not null and not empty.</returns>
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+            => value is IEnumerable list && list.GetEnumerator().MoveNext();
```

Comment:
Should the converter raise "ArgumentException" if the provided value couldn't be cast to IEnumerable?

Reply:
I'm not sure, to be honest. It feels like something as simple as a converter shouldn't crash someone's code if we can't quite manage to convert it. Not sure what other people think about this one? @Depechie @jfversluis @roubachof @bartlannoeye

Reply:
ahah! the classic silently fails vs crash in prod dilemma!

Reply:
In a ideal world we should write something in the error output to not just silently fail.

Reply:
If we don’t want to crash we also need to change the inverse bool converter because it does an implicit type casting instead of using the AS keyword. So the current version will also crash there if no bool was supplied... with AS you get null.

But sure what I like best to be honest :/ if it does not crash you can have a hard time searching for why some XAML is not working as it should...

Reply:
Since this is probably not a discussion we'll manage to figure out on our own I'd say choose one and let the community decide. As per Glenn's last comment, I will change the `InverseBoolConverter` to have everything in line with one another so we can move forward.